src/desdemona/functions/sample_functions.clj

```diff
@@ -1,21 +1,23 @@
 (ns desdemona.functions.sample-functions
-  (:require [clojure
-             [walk :refer [postwalk]]]))
+  (:require
+   [clojure.core.match :refer [match]]))
 
 ;;; Defines functions to be used by the peers. These are located
 ;;; with fully qualified namespaced keywords, such as
-;;; desdemona.functions.sample-functions/format-line
+;;; desdemona.functions.sample-functions/add-message-origin
 
-(defn transform-segment-shape
-  "Recursively restructures a segment {:new-key [paths...]}"
-  [paths segment]
-  (try (let [f (fn [[k v]]
-                 (if (vector? v)
-                   [k (get-in segment v)]
-                   [k v]))]
-         (postwalk (fn [x] (if (map? x) (into {} (map f x)) x)) paths))
-       (catch Exception e
-         segment)))
+(defn build-row [segment]
+  {:line (str (name (segment :origin)) ": " (segment "MESSAGE"))})
```

Comment:
The data that syslog-ng can parse is put into the "_parsed" key. So the JSON looks like:

```
{
"_parsed": { "some: "stuff"},
"CRAP": "FROM SYSLOG",
"LIKE": "IP ADDRESSES AND STUFF",
"MESSAGE": "{\"some\": \"stuff\"}"
}
```

But yeah, then we're adding our enrichments into the top-level namespace alongside the syslog stuff.

Something like this?

```
{
"original": {"_parsed", "MESSAGE", etc},
"origin": "falconhose",
etc...
}
```

might be nicer?